model-optimizer/extensions/load/pytorch/loader.py

```diff
@@ -0,0 +1,117 @@
+"""
+ Copyright (C) 2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import logging as log
+
+import torch
+from torch.autograd import Variable
+
+from extensions.load.loader import Loader
+from mo.front.common.register_custom_ops import update_extractors_with_extensions, check_for_duplicates
+from mo.front.extractor import extract_node_attrs
+from mo.front.pytorch.extractor import pytorch_op_extractor, pytorch_op_extractors
+from mo.graph.graph import Graph
+
+tensors_map = {}
+layer_id = 0
+
+class PyTorchLoader(Loader):
+    enabled = True
+
+    def load(self, graph: Graph):
+        argv = graph.graph['cmd_params']
+        graph.graph['fw'] = 'pytorch'
+        graph.graph['layout'] = 'NCHW'
+
+        update_extractors_with_extensions(pytorch_op_extractors)
+
+        # Create a dummy input
+        inp = Variable(torch.randn(list(argv.placeholder_shapes)))
```

Comment:
Not scale-able solution.

After the loading we extract original Placeholder shapes.
In case user defines new shapes, InputCut transformation performs the shape renewal.

But here we need input shapes for the Loader. So we force user to always define them during MO call.

Multi input topologies are not covered.

Reply:
PyTorch loader requires to specify input shape (as in demo) or input tensor (need to think what's better) because we perform forward pass inside.

```python
mo_pytorch.convert(model, input_shape=[1, 3, 227, 227])
```

> But here we need input shapes for the Loader. So we force user to always define them during MO call.

`input_shape` in this case is MO flag